src/main/java/uk/gov/companieshouse/api/accounts/validation/IntangibleAssetsValidator.java

```diff
@@ -0,0 +1,286 @@
+package uk.gov.companieshouse.api.accounts.validation;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import uk.gov.companieshouse.api.accounts.exception.DataException;
+import uk.gov.companieshouse.api.accounts.exception.ServiceException;
+import uk.gov.companieshouse.api.accounts.model.rest.notes.intangible.Amortisation;
+import uk.gov.companieshouse.api.accounts.model.rest.notes.intangible.IntangibleAssets;
+import uk.gov.companieshouse.api.accounts.model.rest.notes.intangible.IntangibleAssetsResource;
+import uk.gov.companieshouse.api.accounts.model.rest.notes.intangible.Cost;
+import uk.gov.companieshouse.api.accounts.model.validation.Errors;
+import uk.gov.companieshouse.api.accounts.service.CompanyService;
+import uk.gov.companieshouse.api.accounts.service.impl.CurrentPeriodService;
+import uk.gov.companieshouse.api.model.transaction.Transaction;
+
+import javax.servlet.http.HttpServletRequest;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+@Component
+public class IntangibleAssetsValidator  extends BaseValidator  {
+
+    private CompanyService companyService;
+
+    @Autowired
+    public IntangibleAssetsValidator(CompanyService companyService) {
+
+        this.companyService = companyService;
+
+    }
+
+    @Value("${incorrect.total}")
+    private String incorrectTotal;
+
+    @Value("${value.required}")
+    private String valueRequired;
+
+    private static final String INTANGIBLE_NOTE = "$.intangible_assets";
+    private static final String COST_AT_PERIOD_START = ".cost.at_period_start";
+    private static final String ADDITIONS = ".cost.additions";
+    private static final String DISPOSALS = ".cost.disposals";
+    private static final String REVALUATIONS = ".cost.revaluations";
+    private static final String TRANSFERS = ".cost.transfers";
+    private static final String COST_AT_PERIOD_END = ".cost.at_period_end";
+    private static final String NET_BOOK_VALUE_CURRENT_PERIOD = ".net_book_value_at_end_of_current_period";
+
+    public Errors validateIntangibleAssets(IntangibleAssets intangibleAssets, Transaction transaction, String companyAccountsId, HttpServletRequest request)
+    throws DataException {
+        Errors errors = new Errors();
+
+        try {
+            boolean isMultipleYearFiler = companyService.isMultipleYearFiler(transaction);
+
+            List<IntangibleSubResource> invalidSubResources = new ArrayList<>();
+            verifySubResourcesAreValid(intangibleAssets, errors, isMultipleYearFiler, invalidSubResources);
+            verifyNoteNotEmpty(intangibleAssets, errors, isMultipleYearFiler);
+            validateSubResourceTotals(intangibleAssets, errors, isMultipleYearFiler, invalidSubResources);
+            if(errors.hasErrors()) {
```

Comment:
This if block can be removed; it doesn't affect program execution

Reply:
Removed as suggested thank you 😊